### PR TITLE
[Feature] Support tinyint type for paimon jni connector

### LINKED ISSUE #29686: paimon catalog need to support tinyint type
## Enhancement
image
image

### PR DESCRIPTION
**Pull request by leoyy0316**
Fixes #29686

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: java-extensions/jni-connector/src/main/java/com/starrocks/jni/connector/OffHeapColumnVector.java [hunk 1]
```diff
@@ -215,6 +215,20 @@ private void putNull(int rowId) {
...
     }
 
+    public int appendByte(byte v) {
+        reserve(elementsAppended + 1);
+        putByte(elementsAppended, v);
+        return elementsAppended++;
+    }
+
+    public void putByte(int rowId, byte value) {
+        Platform.putByte(null, data + (long) rowId, value);
```
Comment (miomiocat) [inline java-extensions/jni-connector/src/main/java/com/starrocks/jni/connector/OffHeapColumnVector.java:225]: The cast `(long)` can be removed?

File: java-extensions/jni-connector/src/main/java/com/starrocks/jni/connector/ColumnType.java [hunk 2]
```diff
@@ -189,7 +191,7 @@ private void parse(StringScanner scanner) {
             }
             break;
             default: {
-                // convert decimal(x,y) to decimal
+                // convert decimal(x,y) to decimalll
```
[the remark this answers is not in the document]
Comment (leoyy0316) [reply java-extensions/jni-connector/src/main/java/com/starrocks/jni/connector/ColumnType.java]: > typo?

done

File: java-extensions/jni-connector/src/main/java/com/starrocks/jni/connector/OffHeapColumnVector.java (hunk 1 above)
Comment (leoyy0316) [reply java-extensions/jni-connector/src/main/java/com/starrocks/jni/connector/OffHeapColumnVector.java]: > The cast `(long)` can be removed?

done

**Review by miomiocat — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|1|1|100.00%|[]|

**Review by caneGuy — APPROVED** [review]

**Comment by miomiocat** [comment]
@Mergify backport branch-3.1

**Pull request merged**